cache_test.go

```diff
@@ -280,6 +268,61 @@ func testCache(t *testing.T, when spec.G, it spec.S) {
 				assertCacheHasLayer(t, testCache, "buildpack.id:layer-2")
 			})
 		})
+
+		when("buildpack API < 0.6", func() {
+			it.Before(func() {
+				exporter.Buildpacks = []buildpack.GroupBuildpack{{ID: "old.buildpack.id", API: "0.5"}}
+			})
+			when("the layers are valid", func() {
+				it.Before(func() {
+					layerFactory.EXPECT().
+						DirLayer(gomock.Any(), gomock.Any()).
+						DoAndReturn(func(id string, dir string) (layers.Layer, error) {
+							return createTestLayer(id, tmpDir)
+						}).AnyTimes()
+
+					layersDir = filepath.Join("testdata", "cacher", "layers")
+				})
+
+				when("there are previously cached layers", func() {
+					when("the SHAs match", func() {
+						it.Before(func() {
+							metadataTemplate := `{
+					"buildpacks": [
+					 {
+					   "key": "old.buildpack.id",
+					   "layers": {
+					     "cache-true-layer": {
+					       "cache": true,
+					       "sha": "%s",
+					       "data": {"old":"data"}
+					     }
+					   }
+					 }
+					]
+					}`
+							initializeCache(t, exporter, &testCache, cacheDir, layersDir, metadataTemplate)
+						})
+						it("sets cache metadata", func() {
+							err := exporter.Cache(layersDir, testCache)
+							h.AssertNil(t, err)
+
+							metadata, err := testCache.RetrieveMetadata()
+							h.AssertNil(t, err)
+
+							t.Log("adds layer shas to metadata")
```

Comment:
I noticed we don't have a `SHA` assertion here - is that intended?

Reply:
Wrong copy-paste. Thanks!

Reply:
Please see https://github.com/buildpacks/lifecycle/pull/537/commits/1f8bae1526197856d67ae94b129c8e87e77bb576.